### ISSUE TITLE
rawcrop branch: format string problems

### ISSUE THREAD

**Issue by agriggio** [Beep6581/RawTherapee#4287]
in several places in `rtgui/edit.cc` invalid `printf` format strings are used (no arguments are given though the string specifies some). For example, at line 83:
```c++
        printf("objectMap(%p) - cleard !\n");
```
Here `%p` demands a `(void *)` argument, but none is given. Bad things are guaranteed to happen :-)
Please consider using `std::cout` instead -- I know it can be more tedious, but it is way safer.

**Comment by Floessie** [comment]
Gladly, you get some hints from the compiler:

```
/home/user/src/rawtherapee/rtgui/edit.cc: In member function ‘void rtedit::ObjectMOBuffer::flush()’:
/home/user/src/rawtherapee/rtgui/edit.cc:83:44: warning: format ‘%p’ expects a matching ‘void*’ argument [-Wformat=]
         printf("objectMap(%p) - cleard !\n");
                                            ^
/home/user/src/rawtherapee/rtgui/edit.cc: In member function ‘void rtedit::ObjectMOBuffer::resize(int, int)’:
/home/user/src/rawtherapee/rtgui/edit.cc:107:52: warning: format ‘%p’ expects a matching ‘void*’ argument [-Wformat=]
                 printf("objectMap(%p) - cleard !\n");
                                                    ^
```

Best,
Flössie

**Comment by agriggio** [comment]
@Floessie that's how I caught them, in fact :-)
But I suppose not all compilers will warn you, otherwise the code wouldn't be there in the first place...

**Comment by Floessie** [comment]
Or maybe someone had his sunglasses on... 😎

**Comment by Hombre57** [comment]
I just added a comment : https://github.com/Beep6581/RawTherapee/commit/0db576ad589ace6f9626b7929a4b3989d7dc3d83#commitcomment-26744310

I talked about that on IRC with @heckflosse and @Beep6581 before @agriggio chimed in. I should have warned earlier.

**Comment by Hombre57** [comment]
Fixed in commit  69f4c9d. In a moment of annoyance, I just fucked the commit message as well 😠 . I'm not sure to be in good condition for coding tonight...

**Comment by Floessie** [comment]
Oops, then sorry for the noise.
